transformations/subject_object_switch/README.md

```diff
@@ -0,0 +1,27 @@
+# Subject Object Switch
+This transformation switches with each other the subject and object of English sentences to generate new sentences with a very high surface similarity but very different meaning. Can be used, for example, for augmenting data for Semantic Similarity English models.
+
+Author name: William Soto
+Author email: [williamsotomartinez@gmail.com](mailto:williamsotomartinez@gmail.com)
+Author Affiliation: [SyNaLP](https://synalp.loria.fr/)([LORIA](https://www.loria.fr/en/))
+
+## What type of a transformation is this?
+If one or both sentences of a pair contain identifiables subject and object it will switch their places, creating new sentences with the different meaning.
+
+## What tasks does it intend to benefit?
+This transformation can be used for tasks like Paraphrase Detection, Paraphrase Generation, Semantic Similarity, and Entailment. It was particularly thought to avoid Semantic Similarity models giving high similarity to oposite sentences.
+
+## What are the limitations of this transformation?
+The transformation is very simple and can only be applied on a limited set of sentences (namely, those with subject and object and a positive target).
+
+## Robustness Evaluation
+
+### Sentence Operation
+| Transformation                   | roberta-base-SST-2   | bert-base-uncased-QQP   | roberta-large-mnli   | roberta-base-imdb   |
+|:---------------------------------|:---------------------|:------------------------|:---------------------|:--------------------|
+| SentenceSubjectObjectSwitch      | 94.0->93.0 ( -1.0)   | 92.0->88.0 ( -4.0)      | 91.0->88.0 ( -3.0)   | 95.0->95.0 (  0.0)  |
+
+### Sentence Pair Operation
+| Transformation                   | paraphrase-xlm-r-multilingual-v1-MSRP | paraphrase-xlm-r-multilingual-v1-PAWS |
+|:---------------------------------|:--------------------------------------|:--------------------------------------|
+| PairSubjectObjectSwitch          | 69.0->30.0 (-39.0)                    | 44.0->24.0 (-20.0)                    |
```

Comment:
Wow! This is an incredibly large difference!

Reply:
Hello @kaustubhdhole and thank you for your review and your work.

> Without looking at the actual subjects and objects, my best guess would be that the changing of subject and object might have had sometimes been an interchange of a noun and pronoun.. it would be interesting to hear from @Sotwi if he has looked at some examples randomly to see why this performance gap exists!

I haven't looked at the specific data in this case, but if I have some time I will definitely give it a look. That being said, I've been working on paraphrase evaluation for the last couple of months and a problem we encountered on models for semantic similarity was that similar inputs with slight changes that affected the meaning (like this one) would obtain high semantic similarity all the same. The PAWS dataset is an attempt to solve that by presenting very similar sentences with high semantic similarity variation. Hopefully this transformation can help produce more of those samples and impact semantic similarity models in the future.